gcovr/gcov.py

```diff
@@ -700,15 +706,21 @@ def select_gcov_files_from_stdout(out, gcov_filter, gcov_exclude, logger):
             logger.verbose_msg("Excluding gcov file {0}", fname)
             continue
 
-        active_files.append(fname)
+        if tempdir:
+            import shutil
+            active_files.append(os.path.join(tempdir, fname))
+            if chdir != tempdir:
+                shutil.copyfile(os.path.join(chdir, fname), active_files[-1])
+        else:
+            active_files.append(os.path.join(chdir, fname))
 
     return active_files, all_files
```

Comment:
`active_files` is the list of files that should be processed, and `all_files` the list of files that should be kept or deleted, depending on the `--keep` flag. Here, `all_files` only receive the `fname` without the `chdir` prefix. I think that is a bug unless `chdir == "."`.

Additionally, the tempdir files are not part of the all_files and will not be cleaned up until the worker exists. And we are copying and not moving the files to the tempdir even when they are guaranteed to be removed later. This isn't incorrect, but it feels inelegant. I don't know what the correct solution would be. Maybe a third return values for files that should be always deleted after processing?

Reply:
I've fixed the all_files bug. 

I think the move vs copy etc... is a bit more of a thorny issue. Windows is throwing file-in-use errors when either moving or deleting. I think the best option would be to copy (as its clear what is happening and avoids a process lock if there is one), keep track of all the files, then to delete after all the threads are complete. I'm working on making Windows happy today and this is what I'll try first.

I've also had to add exception tracking to the threads so their exceptions get raised on the main thread.